frontend/src/views/student/tournament/OpenTournamentView.vue

```diff
@@ -0,0 +1,332 @@
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Start/End Time',
+      value: 'times',
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Number of Questions',
+      value: 'numberOfQuestions',
+      align: 'center',
+      width: '10%'
+    },
+    {
+      text: 'Enrolled',
+      value: 'enrolled',
+      align: 'center',
+      sortable: false,
+      width: '10%'
+    }
+  ];
+
+  async created() {
+    await this.$store.dispatch('loading');
+    try {
+      this.tournaments = await RemoteServices.getOpenedTournamentsForCourseExecution();
+      this.tournaments.sort((a, b) => Tournament.sortById(a, b));
+      let closedTournaments = await RemoteServices.getClosedTournamentsForCourseExecution();
+      closedTournaments.map(t => {
+        if (t.id) this.closedTournamentsId.push(t.id);
+      });
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  async openTournamentDashboard(tournament: Tournament) {
+    this.$emit('close-show-dashboard-dialog', false);
+    if (tournament)
+      await this.$router.push({
+        name: 'tournament-participants'
+      });
+  }
+
+  newTournament() {
+    this.currentTournament = new Tournament();
+    this.createTournamentDialog = true;
+  }
+
+  async onCreateTournament(tournament: Tournament) {
+    this.tournaments.unshift(tournament);
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  onCloseDialog() {
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  openPasswordDialog(tournamentToJoin: Tournament) {
+    this.currentTournament = tournamentToJoin;
+    this.editPasswordDialog = true;
+  }
+
+  onClosePasswordDialog() {
+    this.currentTournament = null;
+    this.editPasswordDialog = false;
+  }
+
+  async joinPrivateTournament(password: string) {
+    this.password = password;
+    if (this.currentTournament)
+      await this.joinPublicTournament(this.currentTournament);
+    this.editPasswordDialog = false;
+    this.currentTournament = null;
+    this.password = '';
+  }
+
+  async joinPublicTournament(tournamentToJoin: Tournament) {
+    const participants = tournamentToJoin.participants;
+    tournamentToJoin.participants = [];
+    try {
+      await RemoteServices.joinTournament(tournamentToJoin, this.password);
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+      tournamentToJoin.participants = participants;
+      return;
+    }
+    tournamentToJoin.enrolled = true;
+    tournamentToJoin.participants = participants;
+  }
+
+  async solveQuiz(tournament: Tournament) {
+    let statementManager: StatementManager = StatementManager.getInstance;
+    const participants = tournament.participants;
+    tournament.participants = [];
+    try {
+      statementManager.statementQuiz = await RemoteServices.solveTournament(
+        tournament
```

Comment:
it seems that it is not necessary to send the tournament, but only the tournament.id

Reply:
Done